aws_lambda_powertools/metrics/provider/cloudwatchemf_provider.py

```diff
@@ -0,0 +1,198 @@
+from __future__ import annotations
+
+import datetime
+import json
+import logging
+import numbers
+from typing import Any, Dict, List, Tuple
+
+from aws_lambda_powertools.metrics.exceptions import (
+    MetricResolutionError,
+    MetricUnitError,
+    MetricValueError,
+    SchemaValidationError,
+)
+from aws_lambda_powertools.metrics.provider import MetricsProvider
+from aws_lambda_powertools.metrics.types import (
+    MetricNameUnitResolution,
+    MetricResolution,
+    MetricSummary,
+    MetricUnit,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class CloudWatchEMFProvider(MetricsProvider):
+    """Class for CloudWatch EMF format provider.
+
+    Creates metrics asynchronously thanks to CloudWatch Embedded Metric Format (EMF).
+    CloudWatch EMF can create up to 100 metrics per EMF object
+    and metrics, dimensions, and namespace created via EMFProvider
+    will adhere to the schema, will be serialized and validated against EMF Schema.
+
+    All provider are used in MetricManager
+
+
+    """
+
+    def __init__(self):
+        self._metric_unit_valid_options = list(MetricUnit.__members__)
+        self._metric_units = [unit.value for unit in MetricUnit]
+        return super().__init__()
+
+    # generic add metrics function
+    def add_metric(self, metrics, name, unit, value, resolution) -> Dict:
```

Comment:
I'm checking the signature of this method and I don't see the `name` variable being used. Is this for compatibility/non-breaking change reasons?